Lib/_pyrepl/windows_console.py

```diff
@@ -407,31 +411,37 @@ def get_event(self, block: bool = True) -> Event | None:
                     continue
                 return None
 
-            key = rec.Event.KeyEvent.uChar.UnicodeChar
+            key_event = rec.Event.KeyEvent
+            raw_key = key = key_event.uChar.UnicodeChar
 
-            if rec.Event.KeyEvent.uChar.UnicodeChar == "\r":
-                # Make enter make unix-like
+            if key == "\r":
+                # Make enter unix-like
                 return Event(evt="key", data="\n", raw=b"\n")
-            elif rec.Event.KeyEvent.wVirtualKeyCode == 8:
+            elif key_event.wVirtualKeyCode == 8:
                 # Turn backspace directly into the command
-                return Event(
-                    evt="key",
-                    data="backspace",
-                    raw=rec.Event.KeyEvent.uChar.UnicodeChar,
-                )
-            elif rec.Event.KeyEvent.uChar.UnicodeChar == "\x00":
+                key = "backspace"
+            elif key == "\x00":
                 # Handle special keys like arrow keys and translate them into the appropriate command
-                code = VK_MAP.get(rec.Event.KeyEvent.wVirtualKeyCode)
-                if code:
-                    return Event(
-                        evt="key", data=code, raw=rec.Event.KeyEvent.uChar.UnicodeChar
-                    )
+                key = VK_MAP.get(key_event.wVirtualKeyCode)
+                if key:
+                    if key in ("left", "right") and key_event.dwControlKeyState & CTRL_ACTIVE:
```

Comment:
```suggestion
                    if key_event.dwControlKeyState & CTRL_ACTIVE:
```
I tried this locally and it works. For example for arrow up the `ctrl up` is passed on as the key. Currently it is ignored, but maybe it is better to just pass whatever we get and let some platform independent part of the code decide what to do with it?

Reply:
Yeah, I like your idea of this code having a standard format, and `Lib\_pyrepl\keymap.py` and `Lib\_pyrepl\unix_eventqueue.py` can be fixed at a later time so they use the same format for all special keys instead of only properly handling `Ctrl`+`←` and `Ctrl`+`→`.